Improve component support documentation (#334)
CERT-Polska/mquery: docs/components.md

- Ursadb (backend database)
- Redis

In a normal situation, there is one web frontend, one shared Redis database, and
for every ursadb instance, there is a group of one or more agents dedicated to
it:

![](./distributed.png)

In most small-to-medium sized deployments there is only one ursadb instance, and
all workers are assigned to it.

More complex configurations are possible, for example, consider this deployment
with internal and external sample index:

![](./distribured-rev.png)

### Web frontend (mquery)

Mquery is a standard web application written in Python (using the
Fastapi framework).

It talks with Redis directly, and schedules tasks for the workers.

For some administrative tasks (like checking ongoing tasks) it also sends requests
to ursadb directly.

### Mquery daemon (agent)

The workhorse of the entire setup. There must be at least one daemon for
every ursadb instance. Daemon's tasks include querying the assigned ursadb
instance for samples and running YARA rules on candidate samples.

### Redis

Is just a shared database used for communication between daemon and mquery.
It is also used as a task queue for jobs scheduled for agents. And it's also

> used for persistent storage of job results. Finally, it's used to store

Reviewer: ```suggestion
used for persistent storage of job results. Finally, it's utilized to store
```

> plugin configuration and job cache for agents. In short, it's pretty overloaded

Reviewer: ```suggestion
plugin configuration and job cache for agents. To sum up, it's pretty overloaded
```
